README.md

```diff
@@ -78,7 +78,9 @@ class { 'haproxy':
   defaults_options => {
     'log'     => 'global',
     'stats'   => 'enable',
-    'option'  => 'redispatch',
+    'option'  => [
+      'redispatch',
+    ],
```

Comment:
I think this *is* the way to make it backwards compatible :) Consider this:

Hash in params.pp:

```puppet
{
  'option' => 'redispatch'
}
```

Hash supplied by the user:

```puppet
{
  'option' => [
    'httplog',
    'dontlognull',
    'http-server-close'
  ]
}
```

When these two hashes are deep-merged the option for `redispatch` will be lost, because in case of conflict the rightmost value will "win". In this case is the user-supplied options array, since the `redispatch` option in the default hash is specified as a plain value, not an array. Arrays are concatenated but a user-supplied array will win over a plain value in the default hash.

Was that somewhat understandable?